Add unique color per severity level

stveit opened the pull request:
## Scope and purpose

Fixes #996

Adds unique coloring per severity level to the badges on the incident list. The colors are the same across themes, theme-specific colors will be discussed in other issues.

light theme before: 
![image](https://github.com/user-attachments/assets/7614e1fe-a964-4225-b8e4-54c913f050fb)

Light theme after:
![image](https://github.com/user-attachments/assets/b1697066-f99f-48f9-a709-c034b9545683)

Dark theme before:
![image](https://github.com/user-attachments/assets/612e0fa6-9311-4444-9eb9-51eda7962bc6)

Dark theme after:
![image](https://github.com/user-attachments/assets/1d6f8b59-386b-4a60-a226-961eb29aa602)



## Contributor Checklist

Every pull request should have this checklist filled out, no matter how small it is.
More information about contributing to Argus can be found in the
[Development docs](https://argus-server.readthedocs.io/en/latest/development.html).

<!-- Add an "X" inside the brackets to confirm -->
<!-- Remove checks that do not apply -->
<!-- Of the checks that do apply: If not checking one or more of the boxes, please explain why below each. -->

* [x] Added a changelog fragment for [towncrier](https://argus-server.readthedocs.io/en/latest/development/howtos/changelog-entry.html)
* [ ] Added/amended tests for new/changed code
* [ ] Added/changed documentation
* [x] Linted/formatted the code with ruff and djLint, easiest by using [pre-commit](https://github.com/Uninett/Argus?tab=readme-ov-file#code-style)
* [x] The first line of the commit message continues the sentence "If applied, this commit will ...", starts with a capital letter, does not end with punctuation and is 50 characters or less long. See our [how-to](https://argus-server.readthedocs.io/en/latest/development/howtos/commit-messages.html)
* [ ] If applicable: Created new issues if this PR does not fix the issue completely/there is further work to be done
* [x] If this results in changes in the UI: Added screenshots of the before and after
* [ ] If this results in changes to the database model: Updated the [ER diagram](https://argus-server.readthedocs.io/en/latest/development/howtos/regenerate-the-ER-diagram.html)

<!-- Make this a draft PR if the content is subject to change, cannot be merged or if it is for initial feedback -->

elfjes commented on src/argus/htmx/tailwindtheme/tailwind.config.js:
```diff
@@ -22,6 +22,16 @@ module.exports = {
         },
         ...{},
+        extend:{
+            colors: {
+                "severity-critical": "red",
+                "severity-high": "orange",
+                "severity-moderate": "yellow",
+                "severity-low": "green",
+                "severity-information": "blue",
+
+            },
```
I like this, we have a very similar setup. I'm wondering about the naming though. Might it be more prudent to use naming based on incident level? `severity-1`, `severity-2` etc

johannaengland replied:
I personally prefer the current one, since it can be hard to remember if 1 or 5 is the most severe one

elfjes replied:
The nice thing about using numbers is that it's more agnostic. We have different names for the various levels, so as long as we remember level -> our name, we don't also have to remember your name -> our name.

This PR currently adds a feature that we also have, and if we can add a few more improvements to it,  we can use your code so that we can ditch or minimize our own implementation.

hmpf replied:
I agree with @elfjes here.

elfjes commented:
In order to have good interoperability with daisyui and alpha values. Some additional work is needed (In tailwind v3, our current version). Please take a look at https://github.com/GEANT/geant-argus/blob/develop/src/geant_argus/geant_argus/templates/tailwind/tailwind.config.js

we have in the tailwind theme part:
```
theme: {
  extend: {
    colors: {
      'incident-major': 'rgba(var(--color-incident-major), <alpha-value>)',
      'incident-minor': 'rgba(var(--color-incident-minor), <alpha-value>)',
      'incident-critical': 'rgba(var(--color-incident-critical), <alpha-value>)',
      'incident-warning': 'var(--fallback-n,oklch(var(--n)))', // neutral
      'incident-clear': 'rgba(var(--color-incident-clear), <alpha-value>)',
    },
  }
}
```
and then for the daisy part, we set the color variables
```js
const incidentColors = {
    '--color-incident-clear': '80, 200, 120', // #50c878, same as success
    '--color-incident-minor': '238, 194, 0', // #eec200, same as warning
    '--color-incident-major': '238, 155, 0', // #ee9b00
    '--color-incident-critical': '240, 67, 67', // #f04343, same as critical
}
...

daisyui: {
  themes: [{
    'argus': {
      ...baseArgusColors,
      ...incidentColors
      },
  }],
},
```

This way we have customizability through daisy, and alpha value support in tailwind. However, please note that this is for tailwind v3. Apparently v4 has this much simplified, so perhaps we should have our eyes on that

hmpf reviewed (changes requested):
Change the names to use the severity numbers to make them agnostic between naming systems.

We could improve the docs slightly (different issue) and talk about severity levels more.

elfjes commented:
Another thing that you may want to think about is to add a `content` color for every severity, so that theme developers can also define a fitting color for the text inside a badge (or button, or something else that has a severity background)

hmpf commented:
So that we don't forget:

- Open and Unacked should be marked with the same color as severity 1/critical
- Closed and Acked should be marked with the same color as severity 5/information
- The color column (will arrive with #1241) should reuse these colors as well. Open+Unacked: 1, Closed+Acked: 5, other: either blank/no color or 4

stveit commented:
I have renamed the colors to use numbers and added a secondary color per severity that is used for text. I have not done any alpha stuff, I think that can be done in a separate issue focused on making the severity colors customizable per theme

hmpf commented:
> This way we have customizability through daisy, and alpha value support in tailwind. However, please note that this is for tailwind v3. Apparently v4 has this much simplified, so perhaps we should have our eyes on that

Tailwind v4 would have to be its own issue and would not happen this sprint, we need finish reaching feature parity and fixing known bugs.

hmpf reviewed (approved):
This needs some follow-up issues but can be merged as is I think.

I guess we should have color variables background and foreground 1 to 5 so that they are themeable that can then be reused everywhere.

Different issue: enforcing colors of status badges and the color status bar., themeable, with docs to show how.
